source/common/stats/allocator_impl.cc

```diff
@@ -38,6 +25,23 @@ const char AllocatorImpl::DecrementToZeroSyncPoint[] = "decrement-zero";
 AllocatorImpl::~AllocatorImpl() {
   ASSERT(counters_.empty());
   ASSERT(gauges_.empty());
+
+  // Move deleted stats into the sets for removeFromSetLockHeld to function.
```

Comment:
s/for removeSetLockHeld/for the ASSERTs in removeFromSetLockHeld/

I think you could actually surround all this in `ifndef NDEBUG` and not go through these loops at all in production.

Reply:
Done